cli/lib/kontena/callbacks/master/deploy/56_set_server_provider_after_deploy.rb

```diff
@@ -0,0 +1,27 @@
+module Kontena
+  module Callbacks
+    class SetServerProviderAfterDeploy < Kontena::Callback
+
+      include Kontena::Cli::Common
+
+      matches_commands 'master create'
+
+      def after
+        return unless command.exit_code == 0
+        return unless config.current_master
+        return unless config.current_master.name == command.result[:name]
+        return unless command.result[:provider]
+
+        require 'shellwords'
+
+        cmd = ['master', 'config', 'set', "server.provider=#{command.result[:provider]}"]
+        spinner "Setting Master configuration server.provider to '#{command.result[:provider]}'" do
+          Retriable.retriable do
+            Kontena.run(cmd.shelljoin)
```

Comment:
It would be nice though to have an initializer for commands, so that you could do something like: `c = Kontena::Cli::FooCommand.new(grid: 'foo', json: true, service: 'bar')` and then something like `c.get_foos` instead of `c.get_foos('foo', 'bar', true)`